src/hotkey/command_executor.cpp

```diff
@@ -362,9 +364,10 @@ bool command_executor::execute_command(const hotkey_command&  cmd, int /*index*/
 
 void command_executor::surrender_game() {
 	if(gui2::show_message(_("Surrender"), _("Do you really want to surrender the game?"), gui2::dialogs::message::yes_no_buttons) != gui2::window::CANCEL) {
-
-	} else {
-
+		playmp_controller* pmc = dynamic_cast<playmp_controller*>(resources::controller);
```

Comment:
Once surrender() is moved to play_controller there is no dynamic_cast need here anymore either.